src/praisonai-agents/praisonaiagents/runtime/registry.py

```
@@ -0,0 +1,159 @@
+        
+        # Discover entry points
+        self._discover_entry_points()
+    
+    def _discover_entry_points(self) -> None:
+        """Discover runtime factories from entry points."""
+        try:
+            for entry_point in iter_entry_points(self._entry_point_group):
+                factory_loader = entry_point.load
+                factory = factory_loader()
+                self._factories[entry_point.name] = factory
+        except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)